src/Fixer/Import/NoUnusedImportsFixer.php

```diff
@@ -86,6 +117,20 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
         }
     }
 
+    protected function createConfigurationDefinition(): FixerConfigurationResolverInterface
+    {
+        return new FixerConfigurationResolver([
+            (new FixerOptionBuilder('comments_match_case', 'Whether to treat comments as case-sensitive.'))
+                ->setAllowedValues([true, false])
+                ->setDefault(false)
+                ->getOption(),
+            (new FixerOptionBuilder('comments_search_annotations_only', 'Whether to analyze only annotations when considering comments.'))
+                ->setAllowedValues([true, false])
+                ->setDefault(false)
+                ->getOption(),
```

Comment:
Initial feedback, after quick look at the code, without digging deeper. Alternatively we could use single option for defining how comments should be processed:

```suggestion
            (new FixerOptionBuilder('comments_match_strategy', 'Defines how comments should be processed.'))
                ->setAllowedValues(['case_insensitive', 'case_sensitive', 'only_annotations'])
                ->setDefault('case_insensitive')
                ->getOption(),
```

That means we don't have explicit casing support, but rather both `case_sensitive` and `only_annotations` should behave as case-sensitive. I am not sure if it's better, but I think we should at least think about it before going further.

Anyway, great to see your proposal, thanks @u01jmg3 😁.